client/users/user-overlays.tsx

```diff
@@ -46,34 +46,35 @@ export function useUserOverlays<E extends HTMLElement = HTMLElement>({
     profileAnchorY,
     clickableElemRef.current ?? null,
   )
-  const [profileOverlayOpen, setProfileOverlayOpen] = useState(false)
+  const [profileOverlayOpen, openProfileOverlay, closeProfileOverlay] = usePopoverController()
+  const [contextMenuOpen, openContextMenu, closeContextMenu] = usePopoverController()
   const [contextMenuEl, setContextMenuEl] = useState<Element | null>(null)
   const [contextMenuAnchorX, setContextMenuAnchorX] = useState(0)
   const [contextMenuAnchorY, setContextMenuAnchorY] = useState(0)
-  const contextMenuOpen = Boolean(contextMenuEl)
 
-  const onOpenProfileOverlay = useCallback(() => {
-    setProfileOverlayOpen(true)
-  }, [])
-  const onCloseProfileOverlay = useCallback(() => {
-    setProfileOverlayOpen(false)
-  }, [])
   const onClick = useCallback(
-    (e?: React.MouseEvent) => {
-      if (!e || !filterClick || !filterClick(userId, e)) {
-        onOpenProfileOverlay()
+    (e: React.MouseEvent) => {
```

Comment:
Wasn't this optional because not all of these were triggered through an actual click? I feel like you're breaking something with this change

Reply:
Hmm, not sure. I checked all the instances where we use this callback and I couldn't find anything like that.